public/app/features/explore/utils/supplementaryQueries_legacy.test.ts

```diff
@@ -0,0 +1,361 @@
+import { flatten } from 'lodash';
+import { from, Observable } from 'rxjs';
+
+import {
+  DataFrame,
+  DataQueryRequest,
+  DataQueryResponse,
+  DataSourceApi,
+  DataSourceWithSupplementaryQueriesSupport,
+  FieldType,
+  LoadingState,
+  LogLevel,
+  LogsVolumeType,
+  MutableDataFrame,
+  SupplementaryQueryType,
+  SupplementaryQueryOptions,
+  toDataFrame,
+} from '@grafana/data';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { DataQuery } from '@grafana/schema';
+
+import { MockDataSourceApi } from '../../../../test/mocks/datasource_srv';
+import { MockDataQueryRequest, MockQuery } from '../../../../test/mocks/query';
+import { ExplorePanelData } from '../../../types';
+import { mockExplorePanelData } from '../__mocks__/data';
+
+import { getSupplementaryQueryProvider } from './supplementaryQueries';
+
+class MockDataSourceWithSupplementaryQuerySupport
```

Comment:
I would add a comment that this whole file should be removed when we remove support for `getDataProvider` in `DataSourceWithSupplementaryQueriesSupport`

Reply:
Done